sdk/servicebus/service-bus/test/atomManagement.spec.ts

```diff
@@ -47,364 +48,444 @@ const alwaysBeDeletedSubscription = "alwaysbedeletedsubscription";
+    alwaysBeExistingEntity: alwaysBeExistingQueue,
+    alwaysBeDeletedEntity: alwaysBeDeletedQueue,
+    parentTopicName: undefined,
+    parentSubscriptionName: undefined
+  },
+  {
+    entityType: EntityType.TOPIC,
+    alwaysBeExistingEntity: alwaysBeExistingTopic,
+    alwaysBeDeletedEntity: alwaysBeDeletedTopic
+  },
+  {
+    entityType: EntityType.SUBSCRIPTION,
+    alwaysBeExistingEntity: alwaysBeExistingSubscription,
+    alwaysBeDeletedEntity: alwaysBeDeletedSubscription,
+    parentTopicName: alwaysBeExistingTopic
+  },
+  {
+    entityType: EntityType.RULE,
+    alwaysBeExistingEntity: alwaysBeExistingRule,
+    alwaysBeDeletedEntity: alwaysBeDeletedRule,
+    parentTopicName: alwaysBeExistingTopic,
+    parentSubscriptionName: alwaysBeExistingSubscription
+  }
+].forEach((testCase) => {
+  describe(`Atom management - Basic CRUD on "${testCase.entityType}" entities #RunInBrowser`, function(): void {
+    it(`Lists on non-existing entities for type ${testCase.entityType} returns empty array successfully`, async () => {
+      let response;
+      try {
+        await deleteEntity(
+          testCase.entityType,
+          testCase.alwaysBeExistingEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        log.httpAtomXml("Ignoring clean up step");
+      }
 
-        should.equal(response.statusCode, 409, "Error must not be undefined");
-      });
+      response = await listEntities(
+        testCase.entityType,
+        testCase.parentTopicName,
+        testCase.parentSubscriptionName
+      );
 
-      it(`Lists available ${entityType} entities successfully`, async () => {
-        let response;
+      try {
+        await createEntity(
+          testCase.entityType,
+          testCase.alwaysBeExistingEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        log.httpAtomXml("Ignoring clean up step");
+      }
 
-        switch (entityType) {
-          case EntityType.QUEUE:
-          case EntityType.TOPIC:
-            response = await listEntities(entityType);
-            break;
+      should.equal(Array.isArray(response), true, "Result must be any array for list requests");
+      should.equal(
+        response.length,
+        testCase.entityType == EntityType.RULE ? 1 : 0,
+        "Result must be an empty array except for rules"
+      );
+    });
 
-          case EntityType.SUBSCRIPTION:
-            response = await listEntities(entityType, alwaysBeExistingTopic);
-            break;
+    it(`Lists on existing entities for type ${testCase.entityType} with top 1 returns the first entity`, async () => {
+      let response;
 
-          case EntityType.RULE:
-            response = await listEntities(
-              entityType,
-              alwaysBeExistingTopic,
-              alwaysBeExistingSubscription
-            );
-            break;
+      try {
```

Comment:
I think it could be useful to remove the try/catch on both of these createEntity calls. If either fails, it would help us know in CI that this was the root cause for the test failing.

Also, you can have a false positive currently if the 2nd createEntity fails, since then there'd only be one result to return.

Reply:
Updated, also see - https://github.com/Azure/azure-sdk-for-js/pull/6112#issuecomment-555716170